File: src/models/site.rs
use anyhow::Result;
use serde::Deserialize;
use std::fs::read_to_string;

#[derive(Clone, Debug, Default, PartialEq, Eq, Deserialize)]
pub struct Project {
	pub href: String,
	pub title: String,
	pub description: String,
}

#[derive(Clone, Debug, Default, PartialEq, Eq, Deserialize)]
pub struct Site {
	pub title: String,
	pub host: String,
	pub author: String,
	pub description: String,
	pub bio: String,
	pub projects: Vec<Project>,
}

impl super::Model {
	pub async fn site(&self) -> Result<Site> {
		let toml =
			read_to_string(self.base_dir.trim_end_matches("/").to_string() + "/content/site.toml")?;
		let site = toml::from_str(&toml)?;

		Ok(site)
	}
}

File: app/models/site.ts
import * as Toml from "@std/toml";
import * as Marked from "marked";

export async function getSite(): Promise<Site> {
  /* create site model */
  const siteContent = await Deno.readTextFile("./content/site.toml");
  const site = Toml.parse(siteContent) as Site;

  for (const project of site.projects) {
    project.description = await Marked.parse(project.description);
  }

  site.bio = await Marked.parse(site.bio);

  return site;
}
